app/src/androidTest/java/com/ncnf/authentication/LoginActivityTest.java

```diff
@@ -219,7 +233,26 @@ public void signUpFragmentUnsuccessfulRegisterTest2(){
         onView(withId(R.id.signUpConfirmPassword)).perform(typeText(validPassword), closeSoftKeyboard());
         onView(withId(R.id.signUpRegisterButton)).perform(click());
 
+        onView(withId(R.id.signUpEmail)).check(matches(hasNoErrorText()));
+        onView(withId(R.id.signUpPassword)).check(matches(hasNoErrorText()));
+        onView(withId(R.id.signUpConfirmPassword)).check(matches(hasNoErrorText()));
+
         verify(mockedAuth).register(anyString(), anyString());
     }
 
+    private static Matcher<View> hasNoErrorText() {
+        return new BoundedMatcher<View, EditText>(EditText.class) {
+
+            @Override
+            public void describeTo(Description description) {
+                description.appendText("has no error text: ");
+            }
+
+            @Override
+            protected boolean matchesSafely(EditText view) {
+                return view.getError() == null;
+            }
+        };
+    }
+
```

Comment:
Wouldn't it more robust to match against an error text ?

Reply:
I only use it to check that the error field of some component is empty so I don't think we need to